src/main/scala/streamer/TemporalAddrGenUnit.scala

```diff
@@ -0,0 +1,202 @@
+package streamer
+
+import chisel3._
+import chisel3.util._
+
+// input and output declaration for temporal address generation unit
+class TemporalAdressGenUnitIO(
+    temporalLoopDim: Int = TemporalAdressGenUnitTestParameters.temporalLoopDim,
+    temporalLoopBoundWidth: Int =
+      TemporalAdressGenUnitTestParameters.temporalLoopBoundWidth,
+    addrWidth: Int = TemporalAdressGenUnitTestParameters.addrWidth
+) extends Bundle {
```

Comment:
Please add a proper docstring here, what are the inputs and outputs, what is their meaning?
Why are they of a certain type ...?